etils/epath/backend_test.py

```diff
@@ -150,6 +150,40 @@ def _test_glob(
   ]
 
 
+def _test_walk(backend: epath.backend.Backend, tmp_path):
+  nested = (tmp_path / 'abc/nested/')
+  nested.mkdir(parents=True)
+  (tmp_path / 'abc/other_nested/').mkdir(parents=True)
+  (tmp_path / 'abc/nested/001').touch()
+  (tmp_path / 'abc/nested/002').touch()
+  (tmp_path / 'abc/nested/003').touch()
+  link_dir =  (tmp_path / 'abc/link_dir').symlink_to(nested)
+
+  last_seen = None
+
+  for path, dirs, files in backend.walk(tmp_path, top_down=False, follow_symlinks=False):
```

Comment:
Rather than many conditions, How about testing

```python
assert list(backend.walk(tmp_path)) == [
    (tmp_path, ['abc'], []),
    ...,
]
```